### PR TITLE
add a test case which check secret's type

### PR DESCRIPTION
**Pull request by WanLinghao**
Improve unit test in ./pkg/serviceaccount/util_test.go:
We add a test case which check secret's type to make sure provided secret is serviceaccount token kind.
```release-note
NONE
```

### CODE REVIEW

File: pkg/serviceaccount/util_test.go [hunk 1]
```diff
@@ -44,6 +44,20 @@ func TestIsServiceAccountToken(t *testing.T) {
 		},
 	}
 
+	secretTypeMistMatch := &v1.Secret{
```
Comment (liggitt) [inline pkg/serviceaccount/util_test.go]: nit: secretTypeMismatch

**Review by liggitt — COMMENTED** [review]
one nit, lgtm otherwise

**Comment by WanLinghao** [comment]
@liggitt  comments addressed, PTAL

**Pull request merged**